src/ErrorHandler.php

```diff
@@ -54,11 +63,33 @@ public function addFormatters(FormatterInterface ...$formatters): self
         return $this;
     }
 
+    /**
+     * Set a custom log callback
+     */
+    public function logCallback(callable $callback): self
+    {
+        $this->logCallback = $callback;
+
+        return $this;
+    }
+
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
         try {
             return $handler->handle($request);
         } catch (Throwable $error) {
+            if ($this->logger) {
+                if ($this->logCallback) {
+                    ($this->logCallback)($this->logger, $error, $request);
+                } else {
+                    $this->logger->critical('Uncaught exception', [
+                        'message' => $error->getMessage(),
+                        'file' => $error->getFile(),
+                        'line' => $error->getLine(),
```

Comment:
Probably I'd include some info about the request: url, method, and uuid (https://github.com/middlewares/uuid).